packages/lesswrong/lib/rateLimits/constants.ts

```diff
@@ -48,114 +49,93 @@ const LW: {POSTS: PostAutoRateLimit[], COMMENTS: CommentAutoRateLimit[]} = {
     }
   ],
   COMMENTS: [ 
     {
       ...timeframe('1 Comments per 1 hours'),
-      last20KarmaThreshold: -1,
-      downvoterCountThreshold: 3,
       appliesToOwnPosts: false,
+      isActive: (user, features) => features?.last20Karma <= -1 && features?.downvoterCount >= 3,
       rateLimitMessage: `Users with -1 or less karma on recent posts/comments can comment once per hour.<br/>${lwDefaultMessage}`
     }, 
   // 3 comments per day rate limits
     {
       ...timeframe('3 Comments per 1 days'),
-      karmaThreshold: 4,
       appliesToOwnPosts: false,
       rateLimitType: "newUserDefault",
+      isActive: user => user.karma < 5,
       rateLimitMessage: `Users with less than 5 karma can write up to 3 comments a day.<br/>${lwDefaultMessage}`,
     }, 
     {
       ...timeframe('3 Comments per 1 days'), // semi-established users can make up to 20 posts/comments without getting upvoted, before hitting a 3/day comment rate limit
-      last20KarmaThreshold: 1, // requires 1 weak upvote from a 1000+ karma user, or two new user upvotes
-      karmaThreshold: 1999, // at 2000+ karma I trust you more to go on long conversations
       appliesToOwnPosts: false,
+      isActive: (user, features) => user.karma < 2000 && features?.last20Karma <= 0,  // requires 1 weak upvote from a 1000+ karma user, or two new user upvotes, but at 2000+ karma I trust you more to go on long conversations
       rateLimitMessage: `You've recently posted a lot without getting upvoted. Users are limited to 3 comments/day unless their last ${RECENT_CONTENT_COUNT} posts/comments have at least 2+ net-karma.<br/>${lwDefaultMessage}`,
     }, 
   // 1 comment per day rate limits
     {
       ...timeframe('1 Comments per 1 days'),
-      karmaThreshold: -1,
       appliesToOwnPosts: false,
+      isActive: user => user.karma <= -3,
       rateLimitMessage: `Users with -3 or less karma can write up to 1 comment per day.<br/>${lwDefaultMessage}`
     }, 
     {
       ...timeframe('1 Comments per 1 days'),
-      last20KarmaThreshold: -5,
-      karmaThreshold: 1999, // at 2000+ karma, I think your downvotes are more likely to be from people who disagree with you, rather than from people who think you're a troll
-      downvoterCountThreshold: 4,
-      appliesToOwnPosts: false,
-      rateLimitMessage: `Users with -5 or less karma on recent posts/comments can write up to 1 comment per day.<br/>${lwDefaultMessage}`
-    }, 
-    {
-      ...timeframe('1 Comments per 1 days'),
-      last20KarmaThreshold: -5,
-      downvoterCountThreshold: 7,
+
       appliesToOwnPosts: false,
+      isActive: (user, features) => features?.last20Karma <= -5 && features?.downvoterCount >= (user.karma < 2000 ? 4 : 7), // at 2000+ karma, I think your downvotes are more likely to be from people who disagree with you, rather than from people who think you're a troll
```

Comment:
too many layers of nested logic, i'd abstract `(user.karma < 2000 ? 4 : 7)` out into its own descriptively-named variable

Reply:
actually disagree!